scripts/globals/spells/bluemagic/pinecone_bomb.lua

```diff
@@ -0,0 +1,66 @@
+-----------------------------------------
+-- Spell: Pinecone Bomb
+-- Additional effect: sleep. Duration of effect varies with TP
+-- Spell cost: 48 MP
+-- Monster Type: Plantoids
+-- Spell Type: Physical (Piercing)
+-- Blue Magic Points: 2
+-- Stat Bonus: STR+1
+-- Level: 36
+-- Casting Time: 2.5 seconds
+-- Recast Time: 26.5 seconds
+-- Skillchain property: Liquefaction (can open Scission or Fusion)
+-- Combos: None
+-----------------------------------------
+
+require("scripts/globals/magic");
+require("scripts/globals/status");
+require("scripts/globals/bluemagic");
+
+-----------------------------------------
+-- onMagicCastingCheck
+-----------------------------------------
+
+function onMagicCastingCheck(caster,target,spell)
+    return 0;
+end;
+
+-----------------------------------------
+-- OnSpellCast
+-----------------------------------------
+
+local resist = duration*applyResistance();
+function onSpellCast(caster,target,spell)
+    local params = {};
+    -- This data should match information on http://wiki.ffxiclopedia.org/wiki/Calculating_Blue_Magic_Damage
+        params.tpmod = TPMOD_DURATION;
+        params.dmgtype = DMGTYPE_PIERCE;
+        params.scattr = SC_LIQUEFACTION;
+        params.numhits = 1;
+        params.multiplier = 2.25;
+        params.tp150 = 2.25;
+        params.tp300 = 2.25;
+        params.azuretp = 2.25;
+        params.duppercap = 37;
+        params.str_wsc = 0.2;
+        params.dex_wsc = 0.0;
+        params.vit_wsc = 0.0;
+        params.agi_wsc = 0.2;
+        params.int_wsc = 0.0;
+        params.mnd_wsc = 0.0;
+        params.chr_wsc = 0.0;
+    damage = BluePhysicalSpell(caster, target, spell, params);
+    damage = BlueFinalAdjustments(caster, target, spell, damage, params);
+
+	if (damage > 0) then
+		local tp = player:getTP();
+		local getBlueEffectDuration = (tp/100 * 5);
+        local pStrAgi = caster:getStat(MOD_STR) + caster:getStat(MOD_AGI);
+        local mStrAgi = target:getStat(MOD_STR) + target:getStat(MOD_AGI);
+        local dStrAgi = (pStrAgi - mStrAgi);
+        local resist = applyResistance(caster,spell,target,dStrAgi,BLUE_SKILL,0);
+		target:delStatusEffect(EFFECT_SLEEP_I);
```

Comment:
We do. But the script really doesn't matter right now because it can still
never function properly until someone changes the point of sleep removal
for spells to be during the script instead of after
On Apr 16, 2015 12:25 PM, "bendangelo" <notifications@github.com> wrote:

> In scripts/globals/spells/bluemagic/pinecone_bomb.lua
> <https://github.com/DarkstarProject/darkstar/pull/1278#discussion_r28536252>
> :
>
> > +        params.vit_wsc = 0.0;
> > +        params.agi_wsc = 0.2;
> > +        params.int_wsc = 0.0;
> > +        params.mnd_wsc = 0.0;
> > +        params.chr_wsc = 0.0;
> > +    damage = BluePhysicalSpell(caster, target, spell, params);
> > +    damage = BlueFinalAdjustments(caster, target, spell, damage, params);
> > +
> > +	if (damage > 0) then
> > +		local tp = player:getTP();
> > +		local getBlueEffectDuration = (tp/100 * 5);
> > +        local pStrAgi = caster:getStat(MOD_STR) + caster:getStat(MOD_AGI);
> > +        local mStrAgi = target:getStat(MOD_STR) + target:getStat(MOD_AGI);
> > +        local dStrAgi = (pStrAgi - mStrAgi);
> > +        local resist = applyResistance(caster,spell,target,dStrAgi,BLUE_SKILL,0);
> > +		target:delStatusEffect(EFFECT_SLEEP_I);
>
> This should remove all sleep effects, lullaby, etc. I thought we had a
> wakeup method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/1278/files#r28536252>.
>